modules/source-bookshelf/lib/base_methods.js

```diff
@@ -29,16 +31,28 @@ exports.addFilter = function (source) {
 };
 
 exports.addCreate = function (source) {
-  source.model.create = function (params) {
+  source.model.create = function (params, toManyRels) {
     return this.forge(params).save(null, {method: 'insert'}).then(function (model) {
-      return this.forge({id:model.id}).fetch();
+      var self = this;
+      return bPromise.all(toManyRels.map(function(rel) {
+        return model.related(rel.name).attach(rel.ids);
+      })).then(function() {
+        return self.forge({id:model.id}).fetch();
+      });
     }.bind(this));
   };
 };
 
 exports.addUpdate = function (source) {
-  source.model.prototype.update = function (params) {
-    delete params.type;
-    return this.save(params, {patch: true, method: 'update'});
+  source.model.prototype.update = function (params, toManyRels) {
+    return this.save(params, {patch: true, method: 'update'}).then(function (model) {
```

Comment:
Could you add something onto `source` which would act as a way to get a transaction object?

Also this the rationale for making the transaction aware bookshelf object in the next version similar to knex, so much easier when it's baked in to the object construction.